stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

```diff
@@ -492,6 +492,29 @@ public struct Unsafe${Mutable}RawBufferPointer
 %  end # mutable
   }
 
+%if Mutable:
+  /// Exchanges the byte values at the specified offsets
+  /// in the memory region
+  ///
+  /// Both parameters must be valid indices, and not
+  /// equal to `endIndex`. Passing the same index as both `i` and `j` has no
+  /// effect.
+  ///
+  /// - Parameters:
+  ///   - i: The index of the first value to swap.
+  ///   - j: The index of the second value to swap.
+  @_inlineable
+  public func swapAt(_ i: Int, _ j: Int)
+  {
+    guard i != j else { return }
+    let pi = (_position! + i)
+    let pj = (_position! + j)
+    let tmp = pi.load(fromByteOffset: 0, as: UInt8.self)
+    pi.copyBytes(from: pj, count: MemoryLayout<UInt8>.size)
```

Comment:
Why not `count: 1`? If `UInt8` is not one byte, then you've broken the semantic guarantees of the method anyway.

Reply:
> Why not count: 1?

True. Using literal `1` would me more consistent with the pointer arithmetic above.

Reply:
I think using an explicitly typed value from `MemoryLayout` is more consistent with the `load` and `storeBytes` methods, which must be typed.